bob-backend/src/pearl/disk_controller.rs

```diff
@@ -712,4 +698,37 @@ impl DiskController {
             .fold(0, |acc, x| async move { acc + x })
             .await
     }
+
+    async fn get_alien_results_unordered<R, RF, F>(&self, op: &Operation, f: F) -> Option<Vec<R>>
+        where
+            RF: Future<Output = Result<R, Error>>,
+            F: Fn(Group) -> RF 
+    {
+        match self.find_groups(op).await {
+            Ok(groups) => {
+                if groups.is_empty() {
+                    return None;
+                }
+                let mut result = vec![];
+                let mut futures = FuturesUnordered::new();
```

Comment:
Have you checked, that `FuturesUnordered` gives any improvement. I see 2 problems here:
1. All aliens are on the same physical disk, so any disk operation becomes a bottleneck, and parallel execution does not bring benefit
2. Most of the time there will be 0 or 1 group in the vector returned from `self.find_groups(op)`